Source/ResponseSerialization.swift

```diff
@@ -463,251 +423,149 @@ extension DownloadRequest {
-    ///
-    /// - returns: A property list object response serializer.
-    public static func propertyListResponseSerializer(
-        options: PropertyListSerialization.ReadOptions = [])
-        -> DownloadResponseSerializer<Any>
-    {
-        return DownloadResponseSerializer { _, response, fileURL, error in
-            guard error == nil else { return .failure(error!) }
+        guard let validData = data, validData.count > 0 else {
+            if let response = response, emptyDataStatusCodes.contains(response.statusCode) {
+                guard let emptyResponse = Empty.response as? T else {
+                    throw AFError.responseSerializationFailed(reason: .invalidEmptyResponse(type: "\(T.self)"))
+                }
 
-            guard let fileURL = fileURL else {
-                return .failure(AFError.responseSerializationFailed(reason: .inputFileNil))
+                return emptyResponse
             }
 
-            do {
-                let data = try Data(contentsOf: fileURL)
-                return Request.serializeResponsePropertyList(options: options, response: response, data: data, error: error)
-            } catch {
-                return .failure(AFError.responseSerializationFailed(reason: .inputFileReadFailed(at: fileURL)))
-            }
+            throw AFError.responseSerializationFailed(reason: .inputDataNilOrZeroLength)
+        }
+
+        do {
+            return try decoder.decode(T.self, from: validData)
+        } catch {
+            throw error
         }
     }
+}
 
+extension DataRequest {
     /// Adds a handler to be called once the request has finished.
     ///
-    /// - parameter options:           The property list reading options. Defaults to `[]`.
-    /// - parameter completionHandler: A closure to be executed once the request has finished.
-    ///
-    /// - returns: The request.
+    /// - Parameters:
+    ///   - queue:             The queue on which the completion handler is dispatched. Defaults to `nil`, which means
+    ///                        the handler is called on `.main`.
+    ///   - decoder:           The decoder to use to decode the response. Defaults to a `JSONDecoder` with default
+    ///                        settings.
+    ///   - completionHandler: A closure to be executed once the request has finished.
+    /// - Returns:             The request.
     @discardableResult
-    public func responsePropertyList(
-        queue: DispatchQueue? = nil,
-        options: PropertyListSerialization.ReadOptions = [],
-        completionHandler: @escaping (DownloadResponse<Any>) -> Void)
-        -> Self
-    {
-        return response(
-            queue: queue,
-            responseSerializer: DownloadRequest.propertyListResponseSerializer(options: options),
-            completionHandler: completionHandler
-        )
+    public func responseJSONDecodable<T: Decodable>(queue: DispatchQueue? = nil,
```

Comment:
Perhaps it would be useful to add a `type` parameter to this function in order to avoid having to specify `DataResponse<...>` in each completion closure.

```
public func responseJSONDecodable<T: Decodable>(type: T.Type = T.self, queue: DispatchQueue? = nil, decoder: JSONDecoder = JSONDecoder(), completionHandler: @escaping (DataResponse<T>) -> Void) -> Self
```

Calling the `response` method would be somewhat simpler:

```
request.responseJSONDecodable(type: HTTPBinResponse.self) { response in
    ...
}
```

Reply:
That doesn't seem to gain you much, if anything, as it's more verbose to pass it as a parameter. Passing it could also prevent proper type inference if you were to pass a closure to this method.  Adding the type to closure parameter is standard Swift usage for any generic closure like this.

Reply:
Good point. 
This is probably more useful as a convenience for synchronous methods where 'T' is the return type, eg:
`func doSomething<T>(type: T.Type = T.self) -> T`
and then either inferring the return value
`let val = doSomething(type: Int.self)`
or the function parameter parameter
`let val: Int = doSomething()`

I guess one could add the proposed convenience in an extension to `DataRequest`